warehouse/query-core/src/test/java/datawave/query/util/DateIndexIteratorTest.java

```diff
@@ -0,0 +1,247 @@
+        withExpectedNextSeekCounts(10, 1);
+        drive("20250401", "20250403", "LOADED");
+
+        // full scan range, restrict to single day in the column qualifier
+        for (String date : List.of("20250401", "20250402", "20250403")) {
+            withStartEndDate(date, date);
+            withExpectedNextSeekCounts(4, 1);
+            drive("20250401", "20250403", "LOADED");
+        }
+    }
+
+    @Test
+    public void testFieldFilter() throws IOException {
+        write("20250401", "LOADED", "20250401", "datatype-a", "FIELD_A", 1);
+        write("20250401", "LOADED", "20250402", "datatype-a", "FIELD_B", 2);
+        write("20250401", "LOADED", "20250403", "datatype-a", "FIELD_C", 3);
+
+        for (String field : List.of("FIELD_A", "FIELD_B", "FIELD_C")) {
+            withExpectedNextSeekCounts(2, 1);
+            withField(field);
+            drive("20250401", "LOADED");
+        }
+    }
+
+    private void write(String shard, String type, String date, String datatype, String field, int... offsets) {
+        Key key = new Key(shard, type, date + "\u0000" + datatype + "\u0000" + field);
+        BitSet bitset = new BitSet();
+        for (int offset : offsets) {
+            bitset.set(offset);
+        }
+        Value value = new Value(bitset.toByteArray());
+        data.put(key, value);
+    }
+
+    private void drive(String date, String type) throws IOException {
+        drive(date, date, type);
+    }
+
+    private void drive(String start, String stop, String type) throws IOException {
+        Range range = createRangeForDate(start, stop);
+        SortedKeyValueIterator<Key,Value> source = new SortedMapIterator(data);
+
+        Collection<ByteSequence> columnFamily = Collections.singleton(new ArrayByteSequence(type.getBytes()));
+
+        StatDateIndexIterator iter = new StatDateIndexIterator();
+        iter.init(source, options, env);
+        iter.seek(range, columnFamily, true);
+
+        while (iter.hasTop()) {
+            Key key = iter.getTopKey();
+            Value value = iter.getTopValue();
+            log.info("k: {}", key.toStringNoTime());
+            iter.next();
+        }
+
+        assertEquals(expectedNextCount, iter.getNextCount(), "expected next count");
+        assertEquals(expectedSeekCount, iter.getSeekCount(), "expected seek count");
+    }
+
+    private Range createRangeForDate(String start, String stop) {
+        return new Range(start, stop + '~');
+    }
+
+    private void withDatatypeFilter(String... types) {
+        options.put(DateIndexIterator.DATATYPE_FILTER, Joiner.on(',').join(types));
+    }
+
+    private void withStartEndDate(String startDate, String endDate) {
+        options.put(DateIndexIterator.MINIMUM_DATE, startDate);
+        options.put(DateIndexIterator.MAXIMUM_DATE, endDate);
+    }
+
+    private void withField(String field) {
+        options.put(DateIndexIterator.FIELD, field);
+    }
+
+    private void withTimeTravel(boolean flag) {
```

Comment:
I noticed that this function is never called. Will you be adding tests cases later that test the time travel aspect of the iterator?